canonical-web-and-design/maas-ui
# refactor(all): remove all references to general.navigation_options websocket method

## Done

- removed all references to general.navigation_options websocket method from both legacy and react apps

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Run the full UI and go to a React page
- Check that `general.navigationOptions` does not exist in the redux store and no action is dispatched to fetch them
- Go to a legacy page and check that no websocket call is made to `general.navigation_options`
- Check that legacy pages still render properly

## Fixes

Fixes #2580
